Doesn't work with repr(C)

Hi

An enum with a repr(C) will fail to compile:

```
diff --git a/tests/test.rs b/tests/test.rs
index e7e9eff..34ea048 100644
--- a/tests/test.rs
+++ b/tests/test.rs
@@ -4,7 +4,7 @@ mod small_prime {
     use super::*;
 
     #[derive(Serialize_repr, Deserialize_repr, PartialEq, Debug)]
-    #[repr(u8)]
+    #[repr(C)]
     enum SmallPrime {
         Two = 2,
         Three = 3,
```

```
   Compiling serde_repr v0.1.6 (/home/elmarco/src/serde-repr)
error[E0412]: cannot find type `C` in this scope
 --> tests/test.rs:7:12
  |
6 |     #[derive(Serialize_repr, Deserialize_repr, PartialEq, Debug)]
  |              -------------- similarly named type parameter `S` defined here
7 |     #[repr(C)]
  |            ^ help: a type parameter with a similar name exists: `S`

error[E0412]: cannot find type `C` in this scope
 --> tests/test.rs:7:12
  |
6 |     #[derive(Serialize_repr, Deserialize_repr, PartialEq, Debug)]
  |                              ---------------- similarly named type parameter `D` defined here
7 |     #[repr(C)]
  |            ^ help: a type parameter with a similar name exists: `D`
```

Is there a workaround or possible solution ? (is it possible to get the underlying chosen int?)

--- a/src/parse.rs
+++ b/src/parse.rs
@@ -86,10 +86,18 @@ impl Parse for Input {
         let mut repr = None;
         for attr in derive_input.attrs {
             if attr.path().is_ident("repr") {
-                if let Meta::List(_) = &attr.meta {
-                    let ty: Ident = attr.parse_args()?;
-                    repr = Some(ty);
-                    break;
+                if let Meta::List(meta) = &attr.meta {
+                    meta.parse_nested_meta(|meta| {
+                        const RECOGNIZED: &[&str] = &[
+                            "u8", "u16", "u32", "u64", "u128", "usize", "i8", "i16", "i32", "i64",
+                            "i128", "isize",
+                        ];
+                        if RECOGNIZED.iter().any(|int| meta.path.is_ident(int)) {
+                            repr = Some(meta.path.get_ident().unwrap().clone());
+                            return Ok(());
+                        }
+                        Err(meta.error("unsupported repr for serde_repr enum"))
+                    })?;
                 }
             }
         }

--- a/tests/ui/repr_c.rs
+++ b/tests/ui/repr_c.rs
@@ -1,0 +1,12 @@
+use serde_repr::Serialize_repr;
+
+#[derive(Serialize_repr)]
+#[repr(C)]
+enum SmallPrime {
+    Two = 2,
+    Three = 3,
+    Five = 5,
+    Seven = 7,
+}
+
+fn main() {}

--- a/tests/ui/repr_c.stderr
+++ b/tests/ui/repr_c.stderr
@@ -1,7 +1,5 @@
-error[E0412]: cannot find type `C` in this scope
+error: unsupported repr for serde_repr enum
  --> tests/ui/repr_c.rs:4:8
   |
-3 | #[derive(Serialize_repr)]
-  |          -------------- similarly named type parameter `S` defined here
 4 | #[repr(C)]
-  |        ^ help: a type parameter with a similar name exists: `S`
+  |        ^

--- a/tests/ui/repr_c.stderr
+++ b/tests/ui/repr_c.stderr
@@ -1,0 +1,7 @@
+error[E0412]: cannot find type `C` in this scope
+ --> tests/ui/repr_c.rs:4:8
+  |
+3 | #[derive(Serialize_repr)]
+  |          -------------- similarly named type parameter `S` defined here
+4 | #[repr(C)]
+  |        ^ help: a type parameter with a similar name exists: `S`

